GameState.cs

```diff
@@ -76,29 +76,102 @@ public CEntInfoV2 GetEntInfoByIndex(int index)
 
         // warning: expensive -  7ms on i5
         // do not call frequently!
-        public IntPtr GetEntityByName(string name)
+        public IntPtr GetEntityByName(string name, bool fixfalse64bit = false)
         {
             const int MAX_ENTS = 2048; // TODO: is portal2's max higher?
 
-            for (int i = 0; i < MAX_ENTS; i++) 
+            for (int i = 0; i < MAX_ENTS; i++)
             {
                 CEntInfoV2 info = this.GetEntInfoByIndex(i);
                 if (info.EntityPtr == IntPtr.Zero)
                     continue;
 
                 IntPtr namePtr;
-                this.GameProcess.ReadPointer(info.EntityPtr + this.GameOffsets.BaseEntityTargetNameOffset, false, out namePtr);
+                // for some reason, for black mesa in nihilanth's fight the entity pointer returns a false 64bit number with the first half being FFFFFFFF...
+                IntPtr entPtr = fixfalse64bit ? (IntPtr)((uint)info.EntityPtr & 0xFFFFFFFF) : info.EntityPtr;
```

Comment:
huge hack, please remove (sorry)
figure out what's going on here or just remove black mesa for now

Reply:
Do you think it might be because m_pEntity in CEntInfo being an int rather than an uint?

Reply:
Actually a test later I think that's the problem

Reply:
pointers are signed though. sounds to me black mesa modified their CEntInfo and m_pEntity is off by a couple bytes?
a raw hex dump of the CEntInfo would be nice

Reply:
The thing is this only happens sometimes, and the only time it's happened is when the pointer goes over 7FFFFFFF
I tried simulating this using a byre array and it broke exactly how it did in the image

Reply:
pointers over 0x7FF... are invalid on 32 bit, either black mesa has a bug or they changed something. feel free to implement the hack as a workaround but only within the blackmesa code

Reply:
Imma try and a get a hexdump if it ever happens again

Reply:
Ok here we go: https://i.imgur.com/vk2DHxV.png
Entity Pointer is highlighted in red

Reply:
Address does point to a valid location, and making m_pEntity unsigned fixes the issue

Reply:
what's the bottom of View -> Memory Regions look like?

Reply:
looks like I was wrong and addresses can use full 32 bit address space if the exe has the "large address aware" flag https://docs.microsoft.com/en-us/cpp/build/reference/largeaddressaware-handle-large-addresses?view=vs-2019

this is what the skyrim "4gb patch" does

Reply:
so yes, m_pEntity should be uint, thanks